Microsoft.DotNet.Interactive.PowerShell/StreamHandler.cs

```diff
@@ -0,0 +1,180 @@
+                PublishStreamRecord(record, _context, _command);
+            }
+        }
+
+        public void WarningDataAdding(object sender, DataAddingEventArgs e)
+        {
+            if(e.ItemAdded is WarningRecord record)
+            {
+                PublishStreamRecord(record, _context, _command);
+            }
+        }
+
+        private void PublishStreamRecord(
+            object output,
+            KernelInvocationContext context,
+            IKernelCommand command)
+        {
+            context.Publish(
+                new DisplayedValueProduced(
+                    output,
+                    command,
+                    FormattedValue.FromObject(output)));
+        }
+
+        private static void SetupRecordFormatters()
+        {
+            // DebugRecord
+            Formatter<DebugRecord>.Register((record, writer) => {
+                PocketView view = PocketViewTags.pre($"DEBUG: {record.Message}");
+                writer.WriteLine(view.ToDisplayString(HtmlFormatter.MimeType));
+            }, HtmlFormatter.MimeType);
+
+            Formatter<DebugRecord>.Register((record, writer) => {
+                writer.WriteLine($"DEBUG: {record.Message}");
+            }, PlainTextFormatter.MimeType);
+
+            // ErrorRecord
+            Formatter<ErrorRecord>.Register((record, writer) => {
+                string result = null;
+                lock(s_powerShellLock)
+                {
+                    var errorDetails = s_powerShell.AddCommand("Microsoft.PowerShell.Utility\\Out-String")
+                        .AddParameter("InputObject", record)
+                        .InvokeAndClearCommands<string>();
+                    result = errorDetails.Single();
+                }
+
+                if (result != null)
+                {
+                    PocketView view = PocketViewTags.pre(result);
+                    writer.WriteLine(view.ToDisplayString(HtmlFormatter.MimeType));
+                }
+            }, HtmlFormatter.MimeType);
+
+            Formatter<ErrorRecord>.Register((record, writer) => {
+                string result = null;
+                lock(s_powerShellLock)
+                {
+                    var errorDetails = s_powerShell.AddCommand("Microsoft.PowerShell.Utility\\Out-String")
+                        .AddParameter("InputObject", record)
+                        .InvokeAndClearCommands<string>();
+                    result = errorDetails.Single();
+                }
+
+                if (result != null)
+                {
+                    writer.WriteLine(result);
+                }
+            }, PlainTextFormatter.MimeType);
+
+            // InformationRecord
+            Formatter<InformationRecord>.Register((record, writer) => {
+                string prefix = (record.Tags.Count == 1 && record.Tags[0] == "__PipelineObject__") ? "OUTPUT:" : "INFORMATION:";
+                PocketView view = PocketViewTags.pre($"{prefix} {record.MessageData}");
```

Comment:
You can add `using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags` at the top of the file so you don't have to fully qualify calls to `pre`, etc.